src/app/loginComponents/onboarding/downloadcliclient/downloadcliclient.component.ts

```diff
@@ -34,26 +41,41 @@ export class DownloadCLIClientComponent implements OnInit {
     this.dsServerURI = Dockstore.API_URI;
     this.isCopied2 = false;
     let apiVersion = 'unreachable';
+    this.alertService.start('Fetching metadata');
     this.gA4GHService.metadataGet().subscribe(
       (resultFromApi: Metadata) => {
         apiVersion = resultFromApi.version;
         this.dockstoreVersion = `${apiVersion}`;
-        this.downloadCli = `https://github.com/dockstore/dockstore/releases/download/${apiVersion}/dockstore`;
-        this.metadataService
-          .getRunnerDependencies(apiVersion, '3', 'cwltool', 'json')
-          .pipe(finalize(() => this.generateMarkdown()))
-          .subscribe(
-            (json: any) => {
-              if (json) {
-                this.cwltoolVersion = json.cwltool;
-              }
-            },
-            (err) => {
-              console.log('Unable to retrieve requirements.txt file.');
+        this.metadataService.getCliVersion().subscribe(
+          (clijson: CLIInfo) => {
+            if (clijson) {
+              this.alertService.simpleSuccess();
+              this.downloadCli = clijson.cliLatestDockstoreScriptDownloadUrl;
             }
-          );
+            this.metadataService
```

Comment:
Instead of nesting this call, you could use `forkJoin()` --  you don't need any data from the CLIInfo response to make the call to get runner dependencies.

Reply:
Now using forkJoin